src/event_handlers.py

```diff
@@ -0,0 +1,140 @@
+# pylint: disable=too-few-public-methods
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Union
+
+from celery import Task
+from loguru import logger
+
+from src.constants import EventEnum, EventType, LabelName
+from src.helpers import get_exception_class
+from src.instrumentation import EventCounter, EventGauge
+
+
+class InvalidTaskEventLabelName(Exception):
+    pass
+
+
+class IEventHandler(ABC):
+    def __init__(self, state):
+        self._state = state
+
+    def __call__(self, event: Dict[str, Any]):
+        self.handle_event(event=event)
+
+    @abstractmethod
+    def handle_event(self, event: Dict[str, Any]):
+        pass
+
+    @staticmethod
+    def _get_label_values_from_task(
+        instrument: Union[EventCounter, EventGauge], task: Task
+    ) -> Dict[str, Any]:
+        labels = {}
+        for labelname in instrument.labelnames:
+            try:
+                value = getattr(task, labelname)
+            except AttributeError as e:
+                raise InvalidTaskEventLabelName(
```

Comment:
I managed to add a label that was not present on all events and it crashed the app so this at least gives the developer a clean message what is wrong.